design-documents/mde-project-structure.md

```diff
@@ -0,0 +1,41 @@
+# Overview
+
+There is a need to have unified approach to structuring the Git-based project while developing MDEs (Magento Developed Extensions), like Page Builder and MSI.
+
+Currently, MSI is developed in a fork of main Magento repository, while Page Builder is developed in a separate repository (like B2B).
+
+## MDE project structure for Git-based installations
+
+MDE project must be structured as Magento B2B:
+ 1. Modules must be placed under `app/code`
+ 1. Each module must have valid `composer.json` with no version specified. Package version must be generated and added to the `composer.json` during release publication
+ 1. No core modules must be committed to the MDE repository
+ 1. Unit and MFTF tests must be distributed between modules (since they are modular). All other tests must be put inside `dev/tests` like it is done in Magento Open Source repository (because these testing frameworks do not support modularity)
```

Comment:
How other tests are run on the current code base of MSI?
It's preferable to not go back to the old-style approach if possible, and keep tests in modules.

Reply:
@maghamed could you please provide some insight into this?

Reply:
In MSI all modules are placed inside corresponding module directories. All phpunit.xml files in mainline already contain configurations to run such tests. As far as I know PageBuilder team also utilize this approach (e.g. https://github.com/magento/magento2/blob/2.3-develop/dev/tests/integration/phpunit.xml.dist#L24)

Reply:
If the fixture is defined in a module, will it be accessible by tests from the other module?